### PR TITLE
Help and support screen

### PR DESCRIPTION
**Pull request by ZoroarkZark**
## Problem/Purpose

This PR is meant to create both the UI and functionality for the help and support screen.

## Proposed Solution

Creating this screen at a high-level involved the creation of a new page, observing the standard of styling inside of the app, and creating a new system for it. 

For the UI, the format follows the standardized format of the other settings screens.
For the functionality, we required the installation of a new package, expo-mail-composer, to provide the mail composer. A new system, using a hook, was made in order to interact with this package, to send feedback/submit questions/report bugs using the built-in app.

## Results

This PR will provide a new help center page, complete with functionality with an email composer package, but without a dedicated sentry log compiler.

photo_2024-06-28_21-25-17
photo_2024-06-28_21-25-30
photo_2024-06-28_21-25-23

## Tickets 

Tickets
https://trello.com/c/IGbGWAuV

### CODE REVIEW

File: settings-boundary/HelpAndSupport.tsx [hunk 1]
```diff
@@ -0,0 +1,278 @@
...
+    bugReported: async () => {
+      Alert.alert(
+        HELP_AND_SUPPORT_ALERTS.reportBugTapped.title,
+        HELP_AND_SUPPORT_ALERTS.reportBugTapped.description,
+        HELP_AND_SUPPORT_ALERTS.reportBugTapped.buttons(
+          async () => {
+            await env
+              .compileLogs()
+              .then(async (URI) => {
+                try {
```
Comment (ZoroarkZark) [inline settings-boundary/HelpAndSupport.tsx:156]: Don't really need a .then/.catch block, attempt try/catch block for less code clutter

File: settings-boundary/HelpAndSupport.tsx [hunk 2]
```diff
@@ -0,0 +1,278 @@
...
+          <SettingsNavigationLinkView
+            title={"Submit Feedback"}
+            onTapped={state.feedbackSubmitted}
+            iconName={"build"}
+            iconBackgroundColor={AppStyles.black}
+          />
+          <SettingsNavigationLinkView
+            title={"Submit Feedback"}
+            onTapped={state.questionSubmitted}
+            iconName={"help-circle"}
```
Comment (ZoroarkZark) [inline settings-boundary/HelpAndSupport.tsx:271]: Use different icons from View Help Center

File: settings-boundary/HelpAndSupport.tsx [hunk 3]
```diff
@@ -0,0 +1,275 @@
...
+            try {
+              const URI = await env.compileLogs()
+              if (URI) {
+                createEmail(
+                  env.composeEmail,
+                  HELP_AND_SUPPORT_EMAILS.bugReported(URI),
+                  "reportBugSuccess",
+                  "reportBugError"
+                )
+              }
```
Comment (mhayes853) [inline settings-boundary/HelpAndSupport.tsx:168]:
```suggestion
              createEmail(
                env.composeEmail,
                HELP_AND_SUPPORT_EMAILS.bugReported(URI),
                "reportBugSuccess",
                "reportBugError"
              )
```

File: settings-boundary/HelpAndSupport.tsx [hunk 4]
```diff
@@ -0,0 +1,275 @@
...
+  const { data: isShowingContactSection } = useQuery(
+    ["isMailComposerAvailable"],
+    async () => await env.isShowingContactSection(),
+    { initialData: true }
+  )
+  const open = useOpenWeblink()
+
+  return {
+    isShowingContactSection,
+    feedbackSubmitted: async () => {
```
Comment (mhayes853) [inline settings-boundary/HelpAndSupport.tsx:145]:
```suggestion
    feedbackSubmitted: () => {
```

File: settings-boundary/HelpAndSupport.tsx [hunk 5]
```diff
@@ -0,0 +1,275 @@
...
+                    env.composeEmail,
+                    HELP_AND_SUPPORT_EMAILS.bugReported(undefined),
+                    "reportBugSuccess",
+                    "reportBugError"
+                  )
+                )
+              )
+            }
+          },
+          async () => {
```
Comment (mhayes853) [inline settings-boundary/HelpAndSupport.tsx:184]:
```suggestion
          () => {
```

File: settings-boundary/HelpAndSupport.tsx [hunk 6]
```diff
@@ -0,0 +1,275 @@
...
+              HELP_AND_SUPPORT_EMAILS.bugReported(undefined),
+              "reportBugSuccess",
+              "reportBugError"
+            )
+          },
+          () => open(COMPILING_LOGS_INFO_URL)
+        )
+      )
+    },
+    questionSubmitted: async () => {
```
Comment (mhayes853) [inline settings-boundary/HelpAndSupport.tsx:196]:
```suggestion
    questionSubmitted: () => {
```

File: settings-boundary/HelpAndSupport.tsx [hunk 7]
```diff
@@ -0,0 +1,275 @@
...
+    isShowingContactSection,
+    feedbackSubmitted: async () => {
+      createEmail(
+        env.composeEmail,
+        HELP_AND_SUPPORT_EMAILS.feedbackSubmitted,
+        "submitFeedbackSuccess",
+        "submitFeedbackError"
+      )
+    },
+    bugReported: async () => {
```
Comment (mhayes853) [inline settings-boundary/HelpAndSupport.tsx:153]:
```suggestion
    bugReported: () => {
```

File: settings-boundary/HelpAndSupport.tsx [hunk 8]
```diff
@@ -0,0 +1,275 @@
...
+                  HELP_AND_SUPPORT_EMAILS.bugReported(URI),
+                  "reportBugSuccess",
+                  "reportBugError"
+                )
+              }
+            } catch {
+              Alert.alert(
+                HELP_AND_SUPPORT_ALERTS.compileLogError.title,
+                HELP_AND_SUPPORT_ALERTS.compileLogError.description,
+                HELP_AND_SUPPORT_ALERTS.compileLogError.buttons(async () =>
```
Comment (mhayes853) [inline settings-boundary/HelpAndSupport.tsx:173]:
```suggestion
                HELP_AND_SUPPORT_ALERTS.compileLogError.buttons(() =>
```

File: settings-boundary/HelpAndSupport.test.tsx [hunk 9]
```diff
@@ -0,0 +1,174 @@
...
+      )
+    })
+
+    test("Unsuccessful submit question flow", async () => {
+      composeEmail.mockRejectedValue(new Error("Bad error"))
+      const { result } = renderUseHelpAndSupportSettings()
+      await waitFor(() =>
+        expect(result.current.isShowingContactSection).toEqual(true)
+      )
+      await act(async () => await result.current.questionSubmitted())
```
Comment (mhayes853) [inline settings-boundary/HelpAndSupport.test.tsx:121]:
```suggestion
      act(() => result.current.questionSubmitted())
```

File: settings-boundary/HelpAndSupport.test.tsx [hunk 10]
```diff
@@ -0,0 +1,174 @@
...
+        HELP_AND_SUPPORT_EMAIL_SUCCESS_ALERTS.reportBugSuccess.description
+      )
+      expect(composeEmail).toHaveBeenCalledWith(
+        HELP_AND_SUPPORT_EMAILS.bugReported(undefined)
+      )
+    })
+
+    test("Successful submit question flow", async () => {
+      const result = await setupSuccessfulFlow()
+      await act(async () => await result.current.questionSubmitted())
```
Comment (mhayes853) [inline settings-boundary/HelpAndSupport.test.tsx:104]:
```suggestion
      act(() => result.current.questionSubmitted())
```

File: settings-boundary/HelpAndSupport.test.tsx [hunk 11]
```diff
@@ -0,0 +1,174 @@
...
+      )
+      expect(composeEmail).toHaveBeenCalledWith(
+        HELP_AND_SUPPORT_EMAILS.bugReported(TEST_COMPILE_LOGS_URI)
+      )
+    })
+
+    test("Successful report bug flow: logs failure, switch to no logs", async () => {
+      compileLogs.mockRejectedValueOnce(new Error("Something went wrong"))
+      const result = await setupSuccessfulFlow()
+      await act(async () => await result.current.bugReported())
```
Comment (mhayes853) [inline settings-boundary/HelpAndSupport.test.tsx:85]:
```suggestion
      act(() => result.current.bugReported())
```

File: settings-boundary/HelpAndSupport.test.tsx [hunk 12]
```diff
@@ -0,0 +1,174 @@
...
+      )
+      expect(composeEmail).toHaveBeenCalledWith(
+        HELP_AND_SUPPORT_EMAILS.bugReported(undefined)
+      )
+    })
+
+    test("Successful report bug flow: logs success", async () => {
+      compileLogs.mockResolvedValueOnce(TEST_COMPILE_LOGS_URI)
+      const result = await setupSuccessfulFlow()
+      await act(async () => await result.current.bugReported())
```
Comment (mhayes853) [inline settings-boundary/HelpAndSupport.test.tsx:71]:
```suggestion
      act(() => result.current.bugReported())
```

File: settings-boundary/HelpAndSupport.test.tsx [hunk 13]
```diff
@@ -0,0 +1,174 @@
...
+      expect(alertPresentationSpy).toHaveBeenCalledWith(
+        HELP_AND_SUPPORT_EMAIL_ERROR_ALERTS.submitFeedbackError.title,
+        HELP_AND_SUPPORT_EMAIL_ERROR_ALERTS.submitFeedbackError.description
+      )
+    })
+
+    test("Successful report bug flow: no logs selected", async () => {
+      compileLogs.mockRejectedValueOnce(new Error("Logs not compiled"))
+      const result = await setupSuccessfulFlow()
+      await act(async () => await result.current.bugReported())
```
Comment (mhayes853) [inline settings-boundary/HelpAndSupport.test.tsx:57]:
```suggestion
      act(() => result.current.bugReported())
```

File: settings-boundary/HelpAndSupport.test.tsx [hunk 14]
```diff
@@ -0,0 +1,174 @@
...
+      )
+
+      expect(composeEmail).toHaveBeenCalledWith(
+        HELP_AND_SUPPORT_EMAILS.feedbackSubmitted
+      )
+    })
+
+    test("Unsuccessful submit feedback flow", async () => {
+      const result = await setupUnsuccessfulFlow()
+      await act(async () => await result.current.feedbackSubmitted())
```
Comment (mhayes853) [inline settings-boundary/HelpAndSupport.test.tsx:47]:
```suggestion
      act(() => result.current.feedbackSubmitted())
```

File: settings-boundary/HelpAndSupport.test.tsx [hunk 15]
```diff
@@ -0,0 +1,174 @@
...
+      const { result } = renderUseHelpAndSupportSettings()
+      expect(result.current.isShowingContactSection).toEqual(true)
+      await waitFor(() =>
+        expect(result.current.isShowingContactSection).toEqual(false)
+      )
+    })
+
+    test("Successful submit feedback flow", async () => {
+      const result = await setupSuccessfulFlow()
+      await act(async () => await result.current.feedbackSubmitted())
```
Comment (mhayes853) [inline settings-boundary/HelpAndSupport.test.tsx:34]:
```suggestion
      act(() => result.current.feedbackSubmitted())
```

File: settings-boundary/HelpAndSupport.test.tsx [hunk 16]
```diff
@@ -0,0 +1,174 @@
...
+        expect(result.current.isShowingContactSection).toEqual(true)
+      )
+      await act(async () => await result.current.questionSubmitted())
+      expect(alertPresentationSpy).toHaveBeenCalledWith(
+        HELP_AND_SUPPORT_EMAIL_ERROR_ALERTS.submitQuestionError.title,
+        HELP_AND_SUPPORT_EMAIL_ERROR_ALERTS.submitQuestionError.description
+      )
+    })
+
+    const setupSuccessfulFlow = async () => {
```
Comment (mhayes853) [inline settings-boundary/HelpAndSupport.test.tsx:128]:
```suggestion
    const renderSuccessfulEmailCompositionFlow = async () => {
```

File: settings-boundary/HelpAndSupport.test.tsx [hunk 17]
```diff
@@ -0,0 +1,174 @@
...
+      composeEmail.mockResolvedValue("success")
+      const { result } = renderUseHelpAndSupportSettings()
+      await waitFor(() =>
+        expect(result.current.isShowingContactSection).toEqual(true)
+      )
+      return result
+    }
+
+    const setupUnsuccessfulFlow = async () => {
```
Comment (mhayes853) [inline settings-boundary/HelpAndSupport.test.tsx:137]:
```suggestion
    const renderUnsuccessfulEmailCompositionFlow = async () => {
```

File: settings-boundary/HelpAndSupport.tsx [hunk 18]
```diff
@@ -0,0 +1,275 @@
...
+      "Something went wrong while submitting your feedback. Please try again."
+  },
+  reportBugTapped: {
+    title: "Attach Logs?",
+    description:
+      "Adding these logs will help make it easier for us to pinpoint the bug you're having.",
+    buttons: (
+      reportWithLogs: () => Promise<void>,
+      reportWithoutLogs: () => Promise<void>,
+      logQuestion: () => void
```
Comment (mhayes853) [inline settings-boundary/HelpAndSupport.tsx:50]:
```suggestion
     openLogsHelpCenter: () => void
```

File: settings-boundary/HelpAndSupport.tsx [hunk 19]
```diff
@@ -0,0 +1,275 @@
...
+        onPress: reportWithLogs
+      },
+      {
+        text: "No",
+        style: "cancel" as const,
+        onPress: reportWithoutLogs
+      },
+      {
+        text: "What is this?",
+        onPress: logQuestion
```
Comment (mhayes853) [inline settings-boundary/HelpAndSupport.tsx:63]:
```suggestion
        onPress: openLogsHelpCenter
```

**Review by mhayes853 — APPROVED** [review]

**Pull request merged**